src/drivers/dw/dma.c

```diff
@@ -802,20 +802,36 @@ static void dw_dma_verify_transfer(struct dma_chan_data *channel,
 	struct dw_dma_chan_data *dw_chan = dma_chan_get_data(channel);
 #if CONFIG_HW_LLI
 	struct dw_lli *lli = platform_dw_dma_lli_get(dw_chan->lli_current);
-
+#if defined __ZEPHYR__
+	int i;
+#endif
 	switch (next->status) {
 	case DMA_CB_STATUS_END:
 		channel->status = COMP_STATE_PREPARE;
 		dma_reg_write(channel->dma, DW_DMA_CHAN_EN,
 			      DW_CHAN_MASK(channel->index));
 		/* fallthrough */
 	default:
+	/* default action is to clear the DONE bit for all LLI making
+	 * sure the cache is coherent between DSP and DMAC.
+	 */
+#if defined __ZEPHYR__
+		dcache_invalidate_region(dw_chan->lli,
+					 sizeof(struct dw_lli) * channel->desc_count);
+
+		for (i = 0; i < channel->desc_count; i++)
+			dw_chan->lli[i].ctrl_hi &= lli->ctrl_hi &= ~DW_CTLH_DONE(1);
```

Comment:
@lgirdwood is this really intended?

Reply:
Probably not the double & - the desired effect is the probably same since we have to clear the DONE bit in all LLI descriptors. Can you send a PR to remove the middle & (as it should be done anyway in the loop).